src/tests/xudt_rce.rs

```diff
@@ -0,0 +1,953 @@
+#![allow(unused_imports)]
+
+// stdlib or 3rd part lib
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+
+// ckb lib
+use blake2b_rs::{Blake2b, Blake2bBuilder};
+use ckb_crypto::secp::Privkey;
+use ckb_error::assert_error_eq;
+use ckb_hash::blake2b_256;
+use ckb_script::DataLoader;
+use ckb_script::{ScriptError, TransactionScriptsVerifier};
+use ckb_types::packed::{Byte, Byte32};
+use ckb_types::{
+    bytes::Bytes,
+    core::{
+        cell::{CellMeta, CellMetaBuilder, ResolvedTransaction},
+        BlockExt, Capacity, DepType, EpochExt, HeaderView, ScriptHashType, TransactionBuilder,
+        TransactionView,
+    },
+    packed::{
+        self, CellDep, CellInput, CellOutput, OutPoint, Script, WitnessArgs, WitnessArgsBuilder,
+    },
+    prelude::*,
+};
+use rand::rngs::ThreadRng;
+use rand::{thread_rng, Rng};
+use sparse_merkle_tree::{
+    blake2b::Blake2bHasher, default_store::DefaultStore, error::Error, traits::Hasher, MerkleProof,
+    SparseMerkleTree, H256,
+};
+
+// internal lib
+use super::{
+    blockchain::Script as ExtensionScript,
+    build_resolved_tx,
+    xudt_rce_mol::RCCellVecBuilder,
+    xudt_rce_mol::ScriptVec as ExtensionScriptVec,
+    xudt_rce_mol::ScriptVecBuilder as ExtensionScriptVecBuilder,
+    xudt_rce_mol::{RCData, RCDataBuilder, RCRuleBuilder},
+    DummyDataLoader, MAX_CYCLES,
+};
+
+const BLAKE2B_KEY: &[u8] = &[];
+const BLAKE2B_LEN: usize = 32;
+const PERSONALIZATION: &[u8] = b"ckb-default-hash";
+
+pub struct CKBBlake2bHasher(Blake2b);
+
+impl Default for CKBBlake2bHasher {
+    fn default() -> Self {
+        let blake2b = Blake2bBuilder::new(BLAKE2B_LEN)
+            .personal(PERSONALIZATION)
+            .key(BLAKE2B_KEY)
+            .build();
+        CKBBlake2bHasher(blake2b)
+    }
+}
+
+impl Hasher for CKBBlake2bHasher {
+    fn write_h256(&mut self, h: &H256) {
+        self.0.update(h.as_slice());
+    }
+    fn finish(self) -> H256 {
+        let mut hash = [0u8; 32];
+        self.0.finalize(&mut hash);
+        hash.into()
+    }
+}
+
+lazy_static! {
+    pub static ref RCE_HASH: [u8; 32] = [
+        1u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+        0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+    ];
```

Comment:
Related to `RCE_HASH` value issue in c contract code.

`uint8_t RCE_HASH[32] = {1};`

Reply:
We can update the document according to the code later.